datacube/model/__init__.py

```diff
@@ -12,6 +12,7 @@
 from collections import OrderedDict
 from collections.abc import Iterable, Iterator, Mapping, Sequence
 from datetime import datetime
+from functools import cached_property
```

Comment:
One of the cached properties in this file is assigned to in `test_virtual.py:86`, and another one in `test_virtual.py:240`. Is that supported by the functools `cached_property`, or could those two places be fixed easily so they populate the cached_property in some other way?

Reply:
Another question is if `get_by_name_unsafe.cache_clear()` & friends are related to `cached_property` and needs to be updated in some way when replacing the implementation?